core/lib/spree/core/engine.rb

```diff
@@ -105,6 +105,12 @@ class Engine < ::Rails::Engine
         ]
       end
 
+      initializer 'spree.promo.register.promotions.shipping_actions', before: :load_config_initializers do |app|
```

Comment:
At the moment we do, unfortunately.  I'm not sure if it's straightforward to combine them either, since you might remove a shipping promotion from `promotions.actions` so that they don't show up in the admin UI, but you'd still want existing orders to apply existing free shipping promotions?

Very open to ideas on this.